src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -270,21 +270,23 @@ void SyncManager::set_logger_factory(SyncClientConfig::LoggerFactory factory) no
     m_config.logger_factory = std::move(factory);
 }
 
-std::unique_ptr<util::Logger> SyncManager::make_logger() const
+const std::shared_ptr<util::Logger>& SyncManager::make_logger() const
```

Comment:
The reason for the `const` is because `App::sync_manager()` a const reference to the SyncManager. I can change it, but I thought this might be const since it's part of the API?

Reply:
In addition, the state is only changed the first time `make_logger` is called. After that, it returns the cached object without changing state. I could initialize the logger earlier (e.g. when `SyncManager::configure()` is called) and then the `const` holds, since it is returning the already cached logger.

Reply:
Updated logger initialization and renamed `make_logger()` to `get_logger()`